Sources/TuistCore/Utils/XCFrameworkMetadataProvider.swift

```diff
@@ -0,0 +1,62 @@
+import Basic
+import Foundation
+import TuistSupport
+
+enum XCFrameworkMetadataProviderError: FatalError, Equatable {
+    
+    case missingRequiredFile(AbsolutePath)
+
+    // MARK: - FatalError
+
+    var description: String {
+        switch self {
+        case let .missingRequiredFile(path):
+            return "The .xcframework at path \(path.pathString) doesn't contain an Info.plist. It's possible that the .xcframework was not generated properly or that got corrupted. Please, double check with the author of the framework."
```

Comment:
👍 much better. I think the easier we can make the errors to be understood by users the better.